prime-router/src/main/kotlin/fhirengine/engine/FHIRTranslator.kt

```diff
@@ -98,6 +109,26 @@ class FHIRTranslator(
         }
     }
 
+    /**
+     * Takes a [reportId] and returns the content of the original message as a ByteArray
+     */
+    internal fun getOriginalMessage(reportId: ReportId, workflowEngine: WorkflowEngine): ReportFile {
```

Comment:
Both of these helpers should be moved somewhere more general so other services can make use of them as well.